test/show.jl

```diff
@@ -94,6 +94,20 @@ end
 @test_repr "(a / b) / (c / d / e)"
 @test_repr "(a == b == c) != (c == d < e)"
 
+# Exponentiation (>= operator_precedence(:^)) and unary operators
+@test_repr "(-1)^a"
+@test_repr "(-2.1)^-1"
+@test_repr "(-x)^a"
+@test_repr "(-a)^-1"
+@test_repr "(!x)↑!a"
+@test_repr "(!x).a"
+@test_repr "(!x)::a"
+
+# Complex
+
+@test iszero(eval(eval(parse(repr(:($(1 + 2im) - $(1 + 2im)))))))
```

Comment:
It's because `repr` returns a string that looks like `":((1 + 2im) - (1 + 2im))"`, which parses into a double-quoted block.  It can be avoided if using `string` instead of `repr`, but I thought it looked better to use `eval(parse(repr(...)))` as in the `@test_repr` macro ... though then it needs two `eval`s. Probably better just to use `string`, then?